src/test/java/com/redislabs/modules/rejson/ClientTest.java

```diff
@@ -196,4 +256,81 @@ public void type1Exception() throws Exception {
         client.set( "foobar", new FooBarObject(), Path.ROOT_PATH);
         client.type( "foobar", new Path(".foo[1]"));
     }
+
+    @Test
+    public void testMultipleGetAtRootPathAllKeysExist() throws Exception {
+        Baz baz1 = new Baz("quuz1", "grault1", "waldo1");
+        Baz baz2 = new Baz("quuz2", "grault2", "waldo2");
+        Qux qux1 = new Qux("quux1", "corge1", "garply1", baz1);
+        Qux qux2 = new Qux("quux2", "corge2", "garply2", baz2);
+
+        client.set("qux1", qux1);
+        client.set("qux2", qux2);
+
+        List<Qux> oneQux = client.mget(Qux.class, "qux1");
+        List<Qux> allQux = client.mget(Qux.class, "qux1", "qux2");
+
+        assertEquals(1, oneQux.size());
+        assertEquals(2, allQux.size());
+
+        assertEquals(qux1, oneQux.get(0));
+
+        Qux testQux1 = allQux.stream() //
+                .filter(q -> q.quux.equals("quux1")) //
+                .findFirst() //
+                .orElseThrow(() -> new NullPointerException(""));
+        Qux testQux2 = allQux.stream() //
+                .filter(q -> q.quux.equals("quux2")) //
+                .findFirst() //
+                .orElseThrow(() -> new NullPointerException(""));
+
+        assertEquals(qux1, testQux1);
+        assertEquals(qux2, testQux2);
+    }
+
+    @Test
+    public void testMultipleGetAtRootPathWithMissingKeys() throws Exception {
+        Baz baz1 = new Baz("quuz1", "grault1", "waldo1");
+        Baz baz2 = new Baz("quuz2", "grault2", "waldo2");
+        Qux qux1 = new Qux("quux1", "corge1", "garply1", baz1);
+        Qux qux2 = new Qux("quux2", "corge2", "garply2", baz2);
+
+        client.set("qux1", qux1);
+        client.set("qux2", qux2);
+
+        List<Qux> allQux = client.mget(Qux.class, "qux1", "qux2", "qux3");
+
+        System.out.println(allQux);
```

Comment:
Use a logging api if necessary. `slf4j` could be available.
```suggestion
```

Reply:
removed, left overs when debugging